rust/kernel/module_param.rs

```diff
@@ -111,7 +111,7 @@ pub trait ModuleParam: core::fmt::Display + core::marker::Sized {
     ///
     /// The `arg` field of `param` must be an instance of `Self`.
     unsafe extern "C" fn free(arg: *mut crate::c_types::c_void) {
-        core::ptr::drop_in_place(arg as *mut Self);
+        unsafe { core::ptr::drop_in_place(arg as *mut Self) };
```

Comment:
Here and elsewhere you may want to move the `;` inside the `unsafe` block. While this doesn't change behavior as all cases are for `()` returning expressions I think, it does look nicer IMHO.

Reply:
Yeah, now that you mention it, I agree.

Does anyone know if there a way to tell `rustfmt` to look for this? Or in Clippy perhaps?

Reply:
I can't find a clippy lint for this. Makes sense to have as a clippy lint though IMO.

Reply:
+1 Opened https://github.com/rust-lang/rust-clippy/issues/7322.